src/components/admins/MessagesTable.jsx

```diff
@@ -1,60 +1,287 @@
+        setMessagesData(response);
+      } catch (error) {
+        setErrorAlertMessages((prevMessages) => [
+          ...prevMessages,
+          "Eroare la actualizarea datelor: " + error,
+        ]);
+      } finally {
+        setIsLoading(false);
+      }
     };
 
     fetchMessages();
   }, []);
 
+  const handleRefreshTable = async () => {
+    setIsLoading(true);
+    setFilteredMessageList([]);
+    const allMessageData = await getAllMessages();
+    setMessageList(allMessageData);
+    setFilteredMessageList(allMessageData);
+    setIsLoading(false);
+  };
+
+  const handleDownloadTableAsCsv = () => {
+    //TODO: Implement this feature
+    // alert("This feature is not implemented yet.");
+
+    const titleKeys = columns
+      //if there are more columns that should not be included in the csv, add a isExportable property in the columns object
+      .filter((property) => property.Header !== "Id")
+      .map((property) => property.accessor);
+    const refinedData = [];
+    refinedData.push(titleKeys);
+    messageList.forEach((item) => {
+      const rowData = titleKeys.map((key) => item[key]);
+      refinedData.push(rowData);
+    });
+
+    let csvContent = "";
+    refinedData.forEach((row) => {
+      csvContent +=
+        row
+          .map((value) => {
+            if (typeof value === "string") {
+              return value.replace(",", "");
+            } else {
+              return value;
+            }
+          })
+          .join(",") + "\n";
+    });
+
+    const now = new Date();
+    const filename = `Mesaje_HC_${now.getDay()}-${now.getMonth()}-${now.getFullYear()}_${now.getHours()}-${now.getMinutes()}`;
+    const blob = new Blob([csvContent], { type: "text/csv;charset=utf-8;" });
+    if (navigator.msSaveBlob) {
+      navigator.msSaveBlob(blob, filename);
+    } else {
+      var link = document.createElement("a");
+      if (link.download !== undefined) {
+        const url = URL.createObjectURL(blob);
+        link.setAttribute("href", url);
+        link.setAttribute("download", filename);
+        link.style.visibility = "hidden";
+        document.body.appendChild(link);
+        link.click();
+        document.body.removeChild(link);
+      }
+    }
+  };
+
+  const columns = useMemo(
+    () => [
+      {
+        Header: "Nume",
+        accessor: "userName",
+        isFilterable: true,
+      },
+      {
+        Header: "Telefon",
+        accessor: "phone",
+        isFilterable: true,
+      },
+      {
+        Header: "Mesaj",
+        accessor: "content",
+        isFilterable: true,
+      },
+      {
+        Header: "Data",
+        accessor: "sentDate",
+        Cell: ({ value }) => new Date(value).toLocaleDateString(),
+        isFilterable: true,
+      },
+      {
+        Header: "Citit",
+        accessor: "isRead",
+        Cell: ({ value }) => (value ? "Citit" : "Necitit"),
+        isFilterable: true,
+      },
+      {
+        Header: "More",
```

Comment:
I think we can remove this "More" column. There are no extra details to be shown here